services/tasks/local.go

```diff
@@ -664,15 +665,16 @@ func (l *local) writeContent(ctx context.Context, mediaType, ref string, r io.Re
 	if err != nil {
 		return nil, err
 	}
-	if err := writer.Commit(ctx, 0, ""); err != nil {
+	err = writer.Commit(ctx, 0, "")
+	if err != nil && !errdefs.IsAlreadyExists(err) {
 		return nil, err
 	}
 	return &types.Descriptor{
 		MediaType:   mediaType,
 		Digest:      writer.Digest().String(),
 		Size:        size,
 		Annotations: make(map[string]string),
-	}, nil
+	}, err
```

Comment:
I'm not a fan of this pattern. If it is already-exist-error, you might need to call `Info` to reconstruct data.

Reply:
My original idea was to let the writerContent caller sense that the data is not actually being written, but rather that it is using the same content that is already there.
However, as a private method, this may not be necessary.